source/server/listener_impl.cc

```diff
@@ -767,11 +775,12 @@ bool ListenerImpl::supportUpdateFilterChain(const envoy::config::listener::v3::L
     return false;
   }
 
-  // Currently we only support TCP filter chain update.
-  if (Network::Utility::protobufAddressSocketType(config_.address()) !=
-          Network::Socket::Type::Stream ||
-      Network::Utility::protobufAddressSocketType(config.address()) !=
-          Network::Socket::Type::Stream) {
+  if (!Runtime::runtimeFeatureEnabled(
```

Comment:
Does the code that you added in `validateFilterChains` need to be replicated here? I forget the control flow. Can you add a comment on why this is OK given the other check?